### ISSUE TITLE
Feature Request: MacOS Sonoma Notifications Widget

### ISSUE THREAD

**Issue by brettshumaker** [wordpress-mobile/WordPress-iOS#22029]
Hello. I just updated my mac to Sonoma and started playing around with the new widgets. I noticed that widgets from iPhone are also able to be placed on the desktop and thought it might be useful for some folks (maybe only me haha) to be able to place a notifications widget so they can be seen at a glance. Again, this might be specific to how much we use p2s here at a8c, but it would be extremely helpful for me to be able to see a list of my notifications at a glance without having to unlock my phone.

Not sure if it's possible for the iOS app to share the data directly with macOS without the iPhone being unlocked or if the widget itself can be responsible for receiving the notifications directly. No idea how any of that works but wanted to suggest it.

Thanks!

**Comment by twstokes** [comment]
That's a neat idea @brettshumaker! Certainly something to research to determine effort level. 

One concern I'd have (without diving deeper) is the refresh-rate of widgets and if they'd fit well with the real-time nature of notifications. Perhaps on desktop they can be more aggressive when fetching new data, versus mobile.

**Comment by brettshumaker** [comment]
> One concern I'd have (without diving deeper) is the refresh-rate of widgets and if they'd fit well with the real-time nature of notifications.

Yeah, I'm not sure about that either. I've never built anything for iOS/macOS before and I'm not sure how the widget gets its data. I wonder if its through a connection to the app itself on the iOS device or if the macOS widget contains its own logic to fetch the data. Not sure if there would be a way for the iOS app to push something to the widget whenever it sends a push notification to iOS.

I've noticed that my Home widget on macOS (I realize that's a native app) updates a device status immediately when I change it from another device. I just added a 3rd party, iOS only widget to both my Mac desktop and my iPhone and they don't seem to be updating very regularly. I suppose it could be another instance of native apps being allowed to be more interactive than 3rd party apps on a new OS feature.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
